src/components/Footer.tsx

```diff
@@ -0,0 +1,76 @@
+import cn from 'classnames';
+import { useMemo } from 'react';
+import { CompletedFilter } from '../types/CompletedFilter';
+import { Todo } from '../types/Todo';
+
+type Props = {
+  todos: Todo[];
+  completedFilter: CompletedFilter;
+  removeCompletedTodos: () => void;
+  setCompletedFilter: (filter: CompletedFilter) => void;
+};
+
+export const Footer: React.FC<Props> = ({
+  todos,
+  completedFilter,
+  removeCompletedTodos,
+  setCompletedFilter,
+}) => {
+  const uncompletedTodosAmount = useMemo(() => {
+    const uncompletedTodos = todos.filter(todo => !todo.completed);
+
+    return uncompletedTodos.length;
+  }, [todos]);
+
+  const hasCompletedTodos = todos
+    .filter(todoItem => todoItem.completed).length > 0;
```

Comment:
Could find this value without one more filtering by `completed`?

Reply:
good point! I use uncompletedTodosAmount for check.
I made these corrections already in the third part